### PR TITLE
First cut at error section

### LINKED ISSUE #229: Error handling and edge cases
It looks like the spec doesn't say how to handle errors and edger cases.
For examples:
- incompatible curves (`keyAgreement` types in DID DOC) between sender and receiver
- invalid data
- attacks (such as https://github.com/decentralized-identity/didcomm-messaging/issues/217)
- the crypto algorithm that sender wants to use is not supported on the receiver side (in case when DID Comm spec is not fully supported on of the sides, or when new crypto is added to the spec).

It would be great if the spec can address some of this edge cases or provide a general strategy what to do in this cases: just terminate communication, send a specific error message, etc.

### PR DESCRIPTION
**Pull request by dhh1128**
Signed-off-by: Daniel Hardman <daniel.hardman@sicpa.com>

Fixes #229

This is not yet ready to merge because it contains a "TO-DO" item. I wanted to discuss before trying to get to closure.

### CODE REVIEW

File: docs/spec-files/problems.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+The optional `comment` field contains localizable, human-friendly text.
+
+The optional `args` field contains args that are associated with the code when it is defined. In this example, `args` apparently lists a DID that cannot be routed.
+
+The optional `cause` field allows the sender to include a stack trace or other background material to explain what triggered the error. This is akin to the `cause` property of Java's `Throwable`. 
+
+The optional `owner` field declares who the reporter of the problem considers to have the responsibility to do something about it. Typically this would be one of the parties in the protocol. If the sender plans to retry, it would be the sender's DID; if the sender wants to recipient to fix the problem, it would be the recipient's DID.
+
+The required `state_outcome` field tells what the sender's state state machine is in, in the parent protocol, after processing this error. In this case, the sender's state has reverted to `null`, meaning it considers the protocol to be aborted.
```
Comment (kdenhartog) [inline docs/spec-files/problems.md:64]: Let's make sure to discuss this. I'm concerned that revealing information about the state machine could be used as a method to reverse engineer or conduct reconnaissance on an implementation. It's not a hard no for me to not include this, but I'm skeptical about over revealing error states which is why I went the error code route in #83

**Comment by dhh1128** [comment]
Decisions from WG call:

* take out `cause`
* make `state_outcome` optional
* make error codes hierarchical to allow generic handlers of them

**Comment by kdenhartog** [comment]
We should add default hierarchical error codes as well, but can leave that to a different PR.

**Comment by dhh1128** [comment]
Updated per suggestions from Kyle and community.

**Review by kdenhartog — APPROVED** [review]

**Comment by kdenhartog** [comment]
@TelegramSam did you mean to merge this one?

**Pull request merged**